source/renderer/app/components/wallet/settings/ChangeSpendingPasswordDialog.js

```diff
@@ -214,19 +214,26 @@ export default class ChangeSpendingPasswordDialog extends Component<Props> {
 
     const newPasswordClasses = classnames(['newPassword', styles.newPassword]);
 
+    const currentPasswordField = form.$('currentPassword');
+    const newPasswordField = form.$('spendingPassword');
+    const repeatedPasswordField = form.$('repeatPassword');
+
+    const formValuesNotSet = isSpendingPasswordSet
+      ? !currentPasswordField.value ||
+        !newPasswordField.value ||
+        !repeatedPasswordField.value
+      : !newPasswordField.value || !repeatedPasswordField.value;
+
```

Comment:
@DeeJayElly Please change the variable name to `canSubmit` or something similar. Check if `isSubmitting`. Also, handle field errors (you can check in form validator) e.g. `!isSubmittings || ! currentPasswordField.error...`
Add brackets especially after `:`
The part after `?` is not totally correct because you will get string and not boolean. Maybe would be better to split that into few clauses depending if the password is set or not

Reply:
Done